sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/SpringSdkMessageCodec.scala

```diff
@@ -54,7 +54,7 @@ private[springsdk] class SpringSdkMessageCodec extends MessageCodec {
   private def lookupTypeHint(value: Any): String =
     lookupTypeHint(value.getClass)
 
-  private def lookupTypeHint(clz: Class[_]): String = {
+  private[kalix] def lookupTypeHint(clz: Class[_]): String = {
```

Comment:
Would it be more expressive to return `Option[String]`? Although I can imagine that changing the signature here must be a bit expensive, coding-wise.
BTW great catch finding the `null` case.

Reply:
In this case, it will never be null, because we will either register a new entry or get one from the cache

Reply:
Would then make sense to remove the `null` check?

Reply:
the null check is about something different :)

Reply:
:) it seems we are not on the same page hehe. Let me give it a last try I don't want to derail your effort. I'm thinking of the signature as `Option[String]` because of the `null` check. When `typeClass` is `null` the method raises an Exception. I thought it could be better to return a `None` instead of the exception. Hence the change of the signature.

Reply:
as discussed on the call, a new signature will now help much, since this this is not expected situation